File: src/main/java/net/thedragonteam/armorplus/potions/base/PotionBase.java
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.potions.base;

import net.minecraft.potion.Potion;
import net.thedragonteam.armorplus.util.Utils;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 **/
public abstract class PotionBase extends Potion {

    public PotionBase(Boolean isBad, int liquidColor, String name) {
        super(isBad, liquidColor);
        this.setPotionName(Utils.setName(name));
        this.setRegistryName(name);
    }
}

File: src/main/java/net/thedragonteam/armorplus/potions/base/PotionBase.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.potions.base

import net.minecraft.potion.Potion

import net.thedragonteam.armorplus.util.Utils.setName

/**
 * ArmorPlus created by sokratis12GR
 * - TheDragonTeam
 */
abstract class PotionBase(isBadEffectIn: Boolean, liquidColorIn: Int, potionName: String) : Potion(isBadEffectIn, liquidColorIn) {

    init {
        this.setPotionName(setName(potionName))
        this.setRegistryName(potionName)
    }
}
